pgrx/src/iter.rs

```diff
@@ -198,10 +250,38 @@ macro_rules! impl_table_iter {
             }
         }
 
+        unsafe impl<$($C),*> ReturnShipping for ($($C,)*)
+        where
+             $($C: ReturnShipping,)*
+             Self: IntoHeapTuple,
+        {
+            type CallRet = Self;
+
+            fn label_ret(self) -> Ret<Self> {
+                Ret::Once(self)
+            }
+
+            unsafe fn box_return(fcinfo: pg_sys::FunctionCallInfo, ret: Ret<Self>) -> pg_sys::Datum {
+                let value = match ret {
+                    Ret::Zero => return empty_srf(fcinfo),
+                    Ret::Once(value) => value,
+                    Ret::Many(iter, value) => {
+                        iter.into_context(fcinfo);
+                        value
+                    }
+                };
+
+                unsafe {
+                    let fcx = deref_fcx(fcinfo);
+                    let heap_tuple = value.into_heap_tuple((*fcx).tuple_desc);
+                    pg_sys::HeapTupleHeaderGetDatum((*heap_tuple).t_data)
+                }
```

Comment:
This is the most awkward thing that actually makes me think "dealbreaker, maybe?" until I restructure it: the tuple types shouldn't have to have magical knowledge of the TupleDesc that was set up for them by TableIterator.

Reply:
I intend to fix this before we land it but I believe otherwise this is ready for review, I don't want to do much more work on this without feedback.